### ISSUE TITLE
Improve exception safety with smart pointers

### ISSUE THREAD

**Issue by elfring** [memtt/malt#9]
Would you like to wrap any pointer data members with the class template “[std::unique_ptr](https://en.wikipedia.org/wiki/Smart_pointer#unique_ptr "Description for the usage of smart pointers")”?

Update candidates:
- [ProfiledCumulValue](https://github.com/memtt/malt/blob/11f2e71dedc024eb66d8d26da26e9e48097d6352/src/lib/valprof/ProfiledCumulValue.hpp#L48)
- [ProfiledValue](https://github.com/memtt/malt/blob/11f2e71dedc024eb66d8d26da26e9e48097d6352/src/lib/valprof/ProfiledValue.hpp#L149)

**Comment by svalat** [comment]
Yes will do, thanks for looking my code, that's really nice. If you find some other stuff, don't limit yourself.

**Comment by svalat** [comment]
Ok, that require a little bit more work as I didn't yet enabled CXX11 on this project to keep portability on old HPC systems. But will look, now it should be fine mostly on all distributions. 

It needs to rewrite the - [STLInternalAllocator](https://github.com/memtt/malt/blob/db18608f37632adfd3972484702ee93a0c47ba24/src/lib/common/STLInternalAllocator.hpp).

Also I looked on the lines. You speak about the dyanmic arrays ? That's more moving them to `std::vector` & `std::array` than using `std::unique_ptr` no ? Otherwise it will messup if I put composed types inside the ProfiledValue class.

**Comment by elfring** [comment]
I suggest to take another look at the implementation detail when [some new and delete operators are used in affected C++ classes](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-newdelete "C++ core guidelines: R.11: Avoid calling new and delete explicitly").
